packages/config/src/json/json.config.ts

```diff
@@ -0,0 +1,221 @@
+      createdAt: Date.now(),
+      updatedAt: Date.now(),
+    };
+  }
+
+  async tileSet(obj: unknown): Promise<ConfigTileSet> {
+    const ts = zTileSetConfig.parse(obj);
+
+    const imageryFetch: Promise<ConfigImagery>[] = [];
+    if (ts.type === TileSetType.Raster) {
+      for (const layer of ts.layers) {
+        if (layer[2193] != null && layer[2193].startsWith('s3://')) {
+          imageryFetch.push(this.loadImagery(layer[2193], Nztm2000QuadTms, layer.name));
+        }
+
+        if (layer[3857] != null && layer[3857].startsWith('s3://')) {
+          imageryFetch.push(this.loadImagery(layer[3857], GoogleTms, layer.name));
+        }
+      }
+    }
+
+    const imagery = await Promise.all(imageryFetch);
+
+    const layers: ConfigLayer[] = [];
+    const tileSet: Partial<ConfigTileSet> = {
+      type: ts.type,
+      id: ts.id,
+      name: Config.unprefix(Config.TileSet.prefix, ts.id),
+      layers,
+    };
+    function updateLayerUri(uri: string | undefined): string | undefined {
+      if (uri == null) return uri;
+      if (uri.startsWith(ConfigPrefix.Imagery)) return uri;
+      const record = imagery.find((f) => f.uri === uri)?.id; ///
+      if (record == null) throw new Error('Unable to find imagery id for uri:' + uri);
+      return record;
+    }
+
+    // Map the configuration sources into imagery ids
+    for (const l of ts.layers) {
+      const layer = { ...l };
+      layers.push(layer);
+
+      if (tileSet.type === TileSetType.Raster) {
+        if (layer[2193]) layer[2193] = updateLayerUri(layer[2193]);
+        if (layer[3857]) layer[3857] = updateLayerUri(layer[3857]);
+      }
+    }
+    if (ts.title) tileSet.title = ts.title;
+    if (ts.description) tileSet.description = ts.description;
+    if (ts.minZoom) tileSet.minZoom = ts.minZoom;
+    if (ts.maxZoom) tileSet.maxZoom = ts.maxZoom;
+    if (ts.background && tileSet.type === TileSetType.Raster) {
+      tileSet.background = parseRgba(ts.background);
+    }
+
+    if (ts.format) {
+      tileSet.format = ts.format as ImageFormat | VectorFormat;
+    } else {
+      tileSet.format = ts.type === TileSetType.Vector ? VectorFormat.MapboxVectorTiles : ImageFormat.Webp;
+    }
+
+    return tileSet as ConfigTileSet;
+  }
+
+  loadImagery(path: string, tileMatrix: TileMatrixSet, name: string): Promise<ConfigImagery> {
+    let existing = this.cache.get(path);
+    if (existing == null) {
+      existing = this._loadImagery(path, tileMatrix, name);
+      this.cache.set(path, existing);
+    }
+    return existing;
+  }
+
+  async _loadImagery(uri: string, tileMatrix: TileMatrixSet, name: string): Promise<ConfigImagery> {
+    this.logger.trace({ uri }, 'FetchImagery');
+
+    // TODO is there a better way of guessing the imagery id & tile matrix?
+    const id = Config.prefix(ConfigPrefix.Imagery, guessIdFromUri(uri));
+
+    const fileList = await fsa.toArray(fsa.list(uri));
+    const tiffFiles = fileList.filter((f) => f.endsWith('.tiff'));
```

Comment:
Do we consider imagery end with `.tif` ?

Reply:
these our COGs so we decide what they end with